app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt

```diff
@@ -1,28 +1,39 @@
 package com.wire.android.ui.home.messagecomposer
 
+import android.content.Context
+import android.net.Uri
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
+import com.wire.android.ui.home.conversations.model.AttachmentBundle
+import com.wire.android.util.DEFAULT_FILE_MIME_TYPE
+import com.wire.android.util.getMimeType
+import com.wire.android.util.orDefault
+import com.wire.android.util.toByteArray
+import java.io.IOException
 
 @Composable
 fun rememberMessageComposerInnerState(
     defaultMessageText: String = "",
-    defaultMessageComposeInputState: MessageComposeInputState = MessageComposeInputState.Enabled
+    defaultMessageComposeInputState: MessageComposeInputState = MessageComposeInputState.Enabled,
+    defaultAttachmentInnerState: AttachmentInnerState = AttachmentInnerState()
 ): MessageComposerInnerState {
 
     return remember {
```

Comment:
I think instead of passing each time context to fun pickAttachment(context: Context, attachmentUri: Uri)  you can move this line defaultAttachmentInnerState: AttachmentInnerState = AttachmentInnerState() away from the rememberMessageComposerInnerState argument and do it inside the body like this 

{
val context = LocalContext.current
val attachmentInnerState = AttachmentInnerState(context)
}

So tthat AttachmentInnerState holds the context, but I am not sure how safe it is to do that way , if the context gets garbaged collected in the mean time and we could have a memory leak here, I do not think honestly as I think there is only single instance of context as this is application context

Reply:
I don't think we should have a problem with that `context`, I like the idea cause now the context it's getting assigned only once and not for every sharing flow